### ISSUE TITLE
Icosehedran needs texcoords

### ISSUE THREAD

**Issue by richardeakin** [cinder/Cinder#590]
See images from Geometry sample:

![screen shot 2014-11-16 at 2 14 56 am](https://cloud.githubusercontent.com/assets/5132127/5060096/c8367824-6d3e-11e4-9626-6e924f212a29.png)
![screen shot 2014-11-16 at 2 15 02 am](https://cloud.githubusercontent.com/assets/5132127/5060097/cec02db6-6d3e-11e4-802e-b996e839d4e7.png)

**Comment by BanTheRewind** [comment]
Icosahedron appears to not have texture coordinates at all. Just ran into a similar issue.

**Comment by paulhoux** [comment]
What kind of texture coordinates would you recommend? I don't think latlong would yield useful results. All I can think of is barycentric (xyz) coordinates for each triangle.

**Comment by paulhoux** [comment]
Or maybe this... (just kidding)
![image](https://cloud.githubusercontent.com/assets/304908/7130856/fe6fcc9e-e244-11e4-92bf-02185b449848.png)

**Comment by BanTheRewind** [comment]
That's what I do, but I don't know that it should be default. But I don't think this is a "bug". More of just something that needs to be worked with on the user end.

**Comment by BanTheRewind** [comment]
not the map ;)




Making each face the same is my preference.

**Comment by richardeakin** [comment]
Fair enough, issue updated. :)  I do think some sort of texcoords should be produced if it can be enabled on `Icosahedron`, as you'd expect to be able to map something like a stone or clay texture to this (something repeating).

**Comment by paulhoux** [comment]
Pasting a nice link to mapping icosahedra here:
http://www.progonos.com/furuti/MapProj/Normal/ProjPoly/projPoly3.html

When creating the Icosahedron code, I left out the texture coords on purpose, by the way. Precisely because there's no perfect solution. 

Best solution would be a Mapper class that you can apply to each of the ```geom::``` primitives. Then have a ```MapperEquirectangular```, ```MapperCylindrical```, etc.

**Comment by paulhoux** [comment]
Ok, I went ahead and added texture coordinates:
afa23b162ef783d83dc44a4799da74c3c4b94db8

I chose to use a method similar to the fancy image above, but instead of having 10 rectangular images for each pair of triangles, I simply use the same coordinates for each pair. Thus, the same rectangular texture is mapped onto the primitive 10 times, once for each pair of triangles. 

I guess in the future we could support vec3 coordinates, where the z-value is an index into a texture array. But for now this will do.

![2015-05-05_212905](https://cloud.githubusercontent.com/assets/304908/7486106/db5f1bb2-f36d-11e4-9806-22442bad4fbc.png)
